app/src/main/java/com/zulip/android/activities/RecyclerMessageAdapter.java

```diff
@@ -434,14 +440,60 @@ private void setUpGravatar(Message message, MessageHolder messageHolder) {
             float px = TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP,
                     35, resources.getDisplayMetrics());
             String url = message.getSender().getAvatarURL() + "&s=" + px;
+
+            // square default avatar drawable
+            final GradientDrawable defaultAvatar = (GradientDrawable) ContextCompat.getDrawable(context, R.drawable.default_avatar);
```

Comment:
@kunall17 This is definitely a much better approach. The app didn't crash on making these changes :D